viewing_party/party.py

```diff
@@ -1,23 +1,173 @@
 # ------------- WAVE 1 --------------------
-
 def create_movie(title, genre, rating):
-    pass
+    if not title or not genre or not rating:
+        return None
+
+    movie ={}
+    movie["title"] = title
+    movie["genre"] = genre
+    movie["rating"] = rating
+
+    return movie
+
+def add_to_watched(user_data, movie):
+    user_data["watched"].append(movie)
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    user_data["watchlist"].append(movie)
+    return user_data
+
+def watch_movie(user_data, title):
+    for item in user_data["watchlist"]:
+        if item["title"] == title:
+            user_data["watchlist"].remove(item)
+            user_data["watched"].append(item)
+    return user_data
+        
+
 
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
 
+def get_watched_avg_rating(user_data):
+    
+    list_of_ratings = [item["rating"] for item in user_data["watched"]]
+
+    # This will account for the situation when the watched list is empty
+    if len(list_of_ratings) == 0:
+        list_of_ratings.append(0)
+    
+    return sum(list_of_ratings)/len(list_of_ratings)
+
+def get_most_watched_genre(user_data):
+ 
+    list_of_genres = [item["genre"] for item in user_data["watched"]]
+
+    # This will account for the situation when the watched list is empty
+    if len(list_of_genres) == 0:
+        return None
+    
+    # Getting the mode of the list of genres
+    popular_genre = max(set(list_of_genres), key = list_of_genres.count)
+    return popular_genre
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
-        
+def get_unique_watched(user_data):
+    # Method that doesn't use list comprehension, and would work if a friend gave a move a different rating than the user 
+    # list_both_watched = []
+    # for i in range(len(user_data["watched"])):
+    #     for j in range(len(user_data["friends"])):
+    #         for k in range(len(user_data["friends"][j]["watched"])):
+    #             if user_data["watched"][i]["title"] == user_data["friends"][j]["watched"][k]["title"] \
+    #                 and user_data["watched"][i]["title"] not in list_both_watched:
+    #                 list_both_watched.append(user_data["watched"][i])
+    # list_user_watched = []
+    # for i in range(len(user_data["watched"])):
+    #     list_user_watched.append(user_data["watched"][i])
+    # 
+    # list_unique_watched = []
+    # for item in list_user_watched:
+    #     if item not in list_both_watched:
+    #         list_unique_watched.append(item)
+
+
+    list_friend_watched = []
+    for i in range(len(user_data["friends"])):
+        for j in range(len((user_data)["friends"][i]["watched"])):
+            list_friend_watched.append(user_data["friends"][i]["watched"][j])

```

Comment:
This works well! As an alternative, can you think of how you could do this without using `range`?